### DISCUSSION TITLE
What replaced x_OnDemandCost from v.03?

### DISCUSSION THREAD

**Discussion by lundejd** [microsoft/finops-toolkit#895] [General]
I have been converting my FinOps toolkit over to FOCUS 1.0 and the 0.4 PBI files and have run into many broken visuals. (To convert, I've had to copy paste using the advanced editor so I'm not referencing ftk_CostDetails anymore. This also wasn't mentioned anywhere that I could find.)

Right now I'm struggling with missing measures such as x_OnDemandCost. This was discussed in the 0.3 documentation extensively but the 0.4 is not only missing all that in the documentation but I don't see any changes mentioned for where it went and what replaced it. I'm sure I'm just not seeing it as I would think others would also run into these issues if not clearly outlined what happened.

What replaced it? I have a measure that is "TotalCommittmentSavings" which is the sum of x_CommitmentCostSavings (also changed name and is not equivalent to what x_CommitmentSavings was so I have to figure that out as well) / x_OnDemandCost. 

And also, is there a way to get the 0.3 FinOps documentation back so I can see those older measures mentioned again? This would help in converting. The documentation had some great wording on what each meant, like price without negotiated discount, price without commitment discounts, etc. 

If I remember right, x_OnDemandCost was one of those that may or may not be working for all users and had an asterisk on it, but it was working for us.

**Comment by flanakin** [comment]
> I have been converting my FinOps toolkit over to FOCUS 1.0 and the 0.4 PBI files and have run into many broken visuals.

See if this helps: [Moving from FOCUS 1.0 preview to FOCUS 1.0](https://techcommunity.microsoft.com/t5/finops-blog/moving-from-focus-1-0-preview-to-focus-1-0/ba-p/4184346). It's not explicitly about updating reports, but the details should be in there. If you think a separate blog post is warranted for updating reports, I can do that as well. I'll also try to remember to do that when 1.1 comes along in November.

More importantly, I'd love to know what you customized. Should any of that come back into the reports we publish or is it org-specific? 

> (To convert, I've had to copy paste using the advanced editor so I'm not referencing ftk_CostDetails anymore. This also wasn't mentioned anywhere that I could find.)

I'm not sure this is exactly what you're looking for, but we have a [Copy queries from a toolkit report doc](https://microsoft.github.io/finops-toolkit/power-bi/setup#-copy-queries-from-a-toolkit-report). It's not specific to 0.3/0.4 however (and may be outdated 😰).

What you need to do depends on what you customized. We can provide guidance, but there's really a broad spectrum of possibilities here. Knowing what you customized would help.

> Right now I'm struggling with missing measures such as x_OnDemandCost. 

This is in the blog post, but off the top of my head, these are the main changes:
- ChargeSubcategory --> ChargeCategory or ChargeClass or CommitmentDiscountStatus, depending on scenario
- Region --> RegionName
- x_OnDemandCost --> ContractedCost
- x_OnDemandUnitPrice --> ContractedUnitPrice

Did you look in the changelog by any chance? I suppose we could've documented it there if that'd help. Another place might be in the migration guide 🤔 Let me know where you looked and we can figure out where it makes sense to be documented.

My apologies for this... I was hoping the blog post would cover the logical changes for report consumers (minor name changes) and totally missed the customization side of this 😓 

> I have a measure that is "TotalCommittmentSavings" which is the sum of x_CommitmentCostSavings (also changed name and is not equivalent to what x_CommitmentSavings was so I have to figure that out as well) / x_OnDemandCost.

We renamed x_CommitmentSavings to x_CommitmentCostSavings in 0.3. We did that or clarity as we also added savings on the price itself. This change is mentioned in the [changelog](https://microsoft.github.io/finops-toolkit/changelog#-v03) under Power BI. That changed in 0.4 to use ContractedCost, but there shouldn't be any functional change. Let me know if I'm missing something.

> And also, is there a way to get the 0.3 FinOps documentation back so I can see those older measures mentioned again? This would help in converting.

Where are you looking for specifically? If you go to the repo code, you can switch the branch to the v0.3 tag to get the state of the repo we released. That's the only option today. We're moving the docs to Microsoft Learn, which does support versioning better, so I'll make sure to factor that into the plans going forward. 

![Screenshot showing where to select a release tag to see the docs for that release](https://github.com/user-attachments/assets/105bde36-e741-4ab3-a36f-0ae7ee58eaa0)

Here's the [0.3 data dictionary](https://github.com/microsoft/finops-toolkit/blob/v0.3/docs/_resources/data-dictionary.md), if that helps. The formulas aren't in there, but we could easily add them, if that's helpful. My only fear is docs getting outdated, which is very easy to happen when things are disconnected.

> If I remember right, x_OnDemandCost was one of those that may or may not be working for all users and had an asterisk on it, but it was working for us.

Correct. But glad to hear you weren't impacted 🙂 (Side note: Some improvements are coming in 0.5, which will be out by EOM for those who are impacted by this. Not a full fix, but a slightly more complete savings calculation. We're hoping to have a full workaround in 0.6 until Cost Management addresses the underlying data issue.)


Thanks for taking the time to share all of this! Hope this helps. If you ever get any ideas on how to improve docs or features, please don't hesitate to create an issue or even submit a PR 😊
